server/v2/core/appmanager/module.go

```diff
@@ -36,6 +38,10 @@ type STFModule[T transaction.Tx] interface {
 	RegisterPostMsgHandler(router PostMsgRouterBuilder)
 }
 
+type UpgradeModule interface {
+	UpgradeBlocker() func(ctx context.Context) error
+}
```

Comment:
What is UpgradeBlocker? How does it relate to PreBlock?

Reply:
meant to replace preblocker as preblocker was a generalised implementation any module could use while we only want to expose this for upgrade, for now. We will refactor how upgrades work in the future

Reply:
I'm sorry, I'm confused. So we have PreBlocker and UpgradeBlocker now.

Reply:
preblocker was renamed to upgradeblocker

Reply:
Ok I see I was renamed/removed now, thanks! I think when I first reviewed this, they both existed, hence my confusion :)